platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java

```diff
@@ -1118,26 +872,71 @@ void loadReconnectState(final SignedState signedState) {
     private void haltRequested(final String reason) {
         logger.error(EXCEPTION.getMarker(), "System halt requested. Reason: {}", reason);
         freezeManager.freezeEventCreation();
-        for (final StoppableThread thread : chatterThreads) {
-            thread.stop();
-        }
-        for (final StoppableThread thread : syncProtocolThreads) {
-            thread.stop();
-        }
+        gossip.stop();
+    }
 
-        final SyncConfig syncConfig = platformContext.getConfiguration().getConfigData(SyncConfig.class);
-        if (syncConfig.syncAsProtocolEnabled()) {
-            gossipHalted.set(true);
+    /**
+     * Build the event creator.
+     */
+    @Nullable
+    private EventCreator buildEventCreator(@NonNull final EventIntake eventIntake) {
+        final ChatterConfig chatterConfig = platformContext.getConfiguration().getConfigData(ChatterConfig.class);
+        if (chatterConfig.useChatter()) {
+            // chatter has a separate event creator in a different thread. having 2 event creators creates the risk
+            // of forking, so a NPE is preferable to a fork
+            return null;
+        } else {
+            return new EventCreator(
+                    this.appVersion,
+                    selfId,
+                    PlatformConstructor.platformSigner(crypto.getKeysAndCerts()),
+                    consensusRef::get,
+                    swirldStateManager.getTransactionPool(),
+                    eventIntake::addEvent,
+                    eventMapper,
+                    eventMapper,
+                    swirldStateManager.getTransactionPool(),
+                    freezeManager::isFreezeStarted,
+                    new EventCreationRules(List.of()));
         }
+    }
 
-        if (syncManager != null) {
-            syncManager.haltRequestedObserver(reason);
+    /**
+     * Build the event linker.
+     */
+    @NonNull
+    private EventLinker buildEventLinker(@NonNull final List<Predicate<EventDescriptor>> isDuplicateChecks) {
```

Comment:
Objects.requireNonNull().  The usage is in one branch and not the other.

Reply:
added